src/main/java/io/kestra/plugin/notion/AbstractNotionTask.java

```diff
@@ -0,0 +1,180 @@
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public abstract class AbstractNotionTask extends NotionConnection implements RunnableTask<AbstractNotionTask.Output> {
+
+    @Schema(
+        title = "Page ID",
+        description = "The unique identifier of the Notion page"
+    )
+    @NotNull
+    protected Property<String> pageId;
+
+    /**
+     * Common output structure for all Notion page operations
+     */
+    @Builder(toBuilder = true)
+    @Getter
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        
+        @Schema(
+            title = "Page ID",
+            description = "The unique identifier of the page"
+        )
+        private String pageId;
+
+        @Schema(
+            title = "Page URL",
+            description = "The URL of the page in Notion"
+        )
+        private String url;
+
+        @Schema(
+            title = "Page Title",
+            description = "The title of the page"
+        )
+        private String title;
+
+        @Schema(
+            title = "Content",
+            description = "The page content in markdown format"
+        )
+        private String content;
+
+        @Schema(
+            title = "Created Time",
+            description = "When the page was created"
+        )
+        private Instant createdTime;
+
+        @Schema(
+            title = "Last Edited Time", 
+            description = "When the page was last edited"
+        )
+        private Instant lastEditedTime;
+
+        @Schema(
+            title = "Archived",
+            description = "Whether the page is archived"
+        )
+        private Boolean archived;
+
+        @Schema(
+            title = "Properties",
+            description = "Additional page properties"
+        )
+        private Map<String, Object> properties;
+
+        @Schema(
+            title = "Storage URI",
+            description = "URI of the stored file containing detailed page information"
+        )
+        private URI uri;
+
+        @Schema(
+            title = "Success",
+            description = "Whether the operation was successful"
+        )
+        private Boolean success;
+
+        @Schema(
+            title = "Message",
+            description = "Operation result message"
+        )
+        private String message;
+    }
+
+    /**
+     * Helper method to extract page title from Notion page properties
+     */
+    protected String extractPageTitle(Map<String, Object> properties) {
+        if (properties == null) {
+            return null;
+        }
+
+        // Notion stores title in different formats, try common patterns
+        Map<String, Object> titleProp = (Map<String, Object>) properties.get("title");
+        if (titleProp == null) {
+            titleProp = (Map<String, Object>) properties.get("Name");
+        }
+        
+        if (titleProp != null && titleProp.get("title") instanceof java.util.List) {
```

Comment:
And use `List` instead of `java.util.List` everywhere

```suggestion
        if (titleProp != null && titleProp.get("title") instanceof List) {
```

Reply:
fixed by https://github.com/kestra-io/plugin-notion/pull/2/commits/ff7f5e7cbcbbd5e5ed3069de53b98f20c091139b